crates/bevy_editor_core/src/selection.rs

```diff
@@ -2,90 +2,136 @@
 
 use bevy::{ecs::entity::Entities, prelude::*};
 
+use crate::utils::DragCancelClick;
+
 /// Editor selection plugin.
 #[derive(Default)]
 pub struct SelectionPlugin;
 
 impl Plugin for SelectionPlugin {
     fn build(&self, app: &mut App) {
-        app.init_resource::<SelectedEntity>()
-            .register_type::<SelectedEntity>()
-            .add_systems(PostUpdate, reset_selected_entity_if_entity_despawned);
+        app.init_resource::<EditorSelection>()
+            .register_type::<EditorSelection>()
+            .add_systems(PostUpdate, remove_entity_from_selection_if_despawned)
+            .add_observer(selection_handler);
+    }
+}
+
+fn selection_handler(
+    mut trigger: On<Pointer<DragCancelClick>>,
+    selectable_query: Query<(), With<Selectable>>,
+    keyboard_input: Res<ButtonInput<KeyCode>>,
+    mut selection: ResMut<EditorSelection>,
+) {
+    if trigger.button != PointerButton::Primary {
+        return;
+    }
+
+    let target = trigger.target();
+    if selectable_query.contains(target) {
+        trigger.propagate(false);
+        let shift = keyboard_input.any_pressed([KeyCode::ShiftLeft, KeyCode::ShiftRight]);
+        if shift {
+            selection.toggle(target);
+        } else {
+            selection.set(target);
+        }
     }
 }
 
-/// The currently selected entity in the scene.
+/// The currently selected entities in the scene.
 #[derive(Resource, Default, Reflect)]
 #[reflect(Resource, Default)]
-pub struct SelectedEntity(pub Option<Entity>);
+pub struct EditorSelection(Vec<Entity>);
```

Comment:
You should use https://dev-docs.bevy.org/bevy/ecs/entity/type.UniqueEntityVec.html instead, which will move these invariants to the type level and allow for safe parallel access in the future if needed.